packages/contracts/contracts/round/RoundImplementation.sol

```diff
@@ -261,56 +297,44 @@ contract RoundImplementation is AccessControlEnumerable, Initializable {
+  /// @notice Update application, round start & end times (only by ROUND_OPERATOR_ROLE)
+  /// @dev Only updates if new time is in the future and current set time is also in the future
+  /// @param newApplicationsStartTime new applicationsStartTime
+  /// @param newApplicationsEndTime new applicationsEndTime
   /// @param newRoundStartTime new roundStartTime
-  function updateRoundStartTime(uint256 newRoundStartTime) external roundHasNotEnded onlyRole(ROUND_OPERATOR_ROLE) {
-    // slither-disable-next-line timestamp
-    require(newRoundStartTime >= block.timestamp, "time has already passed");
-    require(newRoundStartTime >= applicationsStartTime, "is before app start");
-    require(newRoundStartTime < roundEndTime, "is after round end");
-
-    emit RoundStartTimeUpdated(roundStartTime, newRoundStartTime);
-
-    roundStartTime = newRoundStartTime;
-  }
-
-  /// @notice Update roundEndTime (only by ROUND_OPERATOR_ROLE)
   /// @param newRoundEndTime new roundEndTime
-  function updateRoundEndTime(uint256 newRoundEndTime) external roundHasNotEnded onlyRole(ROUND_OPERATOR_ROLE) {
-    // slither-disable-next-line timestamp
-    require(newRoundEndTime >= block.timestamp, "time has already passed");
-    require(newRoundEndTime > roundStartTime, "is before round start");
-    require(newRoundEndTime >= applicationsEndTime, "is before app end");
-
-    emit RoundEndTimeUpdated(roundEndTime, newRoundEndTime);
-
-    roundEndTime = newRoundEndTime;
-  }
-
-  /// @notice Update applicationsStartTime (only by ROUND_OPERATOR_ROLE)
-  /// @param newApplicationsStartTime new applicationsStartTime
-  function updateApplicationsStartTime(uint256 newApplicationsStartTime) external roundHasNotEnded onlyRole(ROUND_OPERATOR_ROLE) {
+  function updateStartAndEndTimes(
+    uint256 newApplicationsStartTime,
+    uint256 newApplicationsEndTime,
+    uint256 newRoundStartTime,
+    uint256 newRoundEndTime
+  ) external roundHasNotEnded onlyRole(ROUND_OPERATOR_ROLE) {
     // slither-disable-next-line timestamp
-    require(newApplicationsStartTime >= block.timestamp, "time has already passed");
-    require(newApplicationsStartTime <= roundStartTime, "is after round start");
-    require(newApplicationsStartTime < applicationsEndTime, "is after app end");
-
-    emit ApplicationsStartTimeUpdated(applicationsStartTime, newApplicationsStartTime);
-
-    applicationsStartTime = newApplicationsStartTime;
-  }
+    require(newApplicationsStartTime < newApplicationsEndTime, "Round: Application end is before application start");
+    require(newRoundStartTime < newRoundEndTime, "Round: Round end is before round start");
+    require(newApplicationsStartTime < newRoundStartTime, "Round: Round start is before application start");
+    require(newApplicationsEndTime < newRoundEndTime, "Round: Round end is before application end");
+
+    if(newApplicationsStartTime > block.timestamp && applicationsStartTime > block.timestamp) {
```

Comment:
```suggestion
    if(newApplicationsStartTime != applicationsStartTime) {
```

Reply:
Maybe we shift the block.timestamp check to one level above and do something like

```
require(newApplicationsStartTime >= block.timestamp
```

this check would be the only block.timestamp check needed for the entire function